### PR TITLE
Additional Waiting for API Server to become ready

### PR DESCRIPTION
**Pull request by rhamitarora**
### Which issue this PR addresses:

https://issues.redhat.com/browse/ARO-4404

### What this PR does / why we need it:

A long discussion is here https://redhat-internal.slack.com/archives/C02ULBRS68M/p1697570801196159?thread_ts=1695641115.974799&cid=C02ULBRS68M

### Test plan for issue:

None

### Is there any documentation that needs to be updated for this PR?

None

### CODE REVIEW

File: pkg/cluster/install.go [hunk 1]
```diff
@@ -343,6 +343,7 @@ func (m *manager) Install(ctx context.Context) error {
 			steps.Action(m.initializeOperatorDeployer), // depends on kube clients
 			steps.Action(m.removeBootstrap),
 			steps.Action(m.removeBootstrapIgnition),
+			steps.Condition(m.apiServersReady, 30*time.Minute, true), // waiting for API Server to become ready
```
Comment (bennerv) [inline pkg/cluster/install.go:346]: Should we put a comment in here stating the apiserver will occasionally be disrupted and the step to configure certificates will fail?  Currently under investigation.

**Review by bennerv — APPROVED** [review]
lgtm - wondering if we should add a comment though.

**Review by Shivkumar13 — APPROVED** [review]
LGTM

File: pkg/cluster/install.go (hunk 1 above)
Comment (rhamitarora) [reply pkg/cluster/install.go]: Comment updated

**Review by Shivkumar13 — APPROVED** [review]

**Review by bennerv — APPROVED** [review]
lgtm

**Pull request merged**